test/extensions/filters/http/rbac/rbac_filter_test.cc

```diff
@@ -54,12 +56,26 @@ class RoleBasedAccessControlFilterTest : public testing::Test {
     ON_CALL(connection_, localAddress()).WillByDefault(ReturnRef(address_));
   }
 
+  void setMetadata() {
+    metadata_.mutable_filter_metadata()->insert(
+        Protobuf::MapPair<Envoy::ProtobufTypes::String, ProtobufWkt::Struct>(
+            HttpFilterNames::get().Rbac, metrics_));
+
+    ON_CALL(req_info_, setDynamicMetadata(_, _))
```

Comment:
`ON_CALL(req_info_, setDynamicMetadata(HttpFilterNames::get().Rbac, _))` then you can remove the if inside of invoke.

Reply:
done.